src/components/SiteDetail.js

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+import summary from '../apis/summary';
+class SiteDetail extends Component {
+  state = {
+    loadedSites: null
+  };
+
+  componentDidMount() {
+    this.loadSiteData();
+  }
+
+  componentDidUpdate() {
+    this.loadSiteData();
+  }
+
+  loadSiteData() {
+    if (this.props.match.params.id) {
+      if (
```

Comment:
The logic here can get a bit confusing. Have you considered using `object destructuring` to tighten it up?

```js
  async loadSiteData() {
    const { id: matchId } = this.props.match.params;
    const { loadedSites } = this.state;



    if (matchId && (!loadedSites || (loadedSites && loadedSites.id !== matchId))) {
      const { data: loadedSites } = await summary.get(`/sites/${matchId}`);
      console.log('Site Detail -> ', loadedSites);
      this.setState({ loadedSites });
    }
  }
```